### DISCUSSION TITLE
Manual output on enter, no linebreak.

### DISCUSSION THREAD

**Discussion by Shuraken007** [junegunn/fzf#3514] [Q&A] [answered]
Hello. 
I'm sending some arguments to fzf (separated by !) and I want only first argument on output after selecting multiple result and hitting `Enter`.

So .. I added smth like this.
```sh
   export FZF_CTRL_T_OPTS="
   -d!
   --with-nth 2
   --bind \"enter:execute(for e in {+1}; do echo \$e; done)+abort\"
   "
```

But problem is, that `echo \$e` don't adding linebreaks, all selected items printed oneline.
As you see, I can't add some postprocess, cause FZF_CTRL_T - havn't such options.

**Comment by LangLangBart** [comment, marked as answer]
When you press the <kbd>⌃ Control</kbd> + <kbd>T</kbd> hotkey, the `fzf-file-widget` widget is executed, which uses the `__fsel` function.

https://github.com/junegunn/fzf/blob/952b6af4454ed55626d78e3845c6b5b640ac831d/shell/key-bindings.zsh#L63-L68

https://github.com/junegunn/fzf/blob/952b6af4454ed55626d78e3845c6b5b640ac831d/shell/key-bindings.zsh#L42-L56

The `echo -n` line outputs the values on the same line. If you would place the `__fsel` in  your `.zshrc` after sourcing `fzf's` `key-bindings.zsh` and remove the `-n` it would print over multiple lines.

```diff
   local item
   eval "$cmd" | FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} --reverse --scheme=path --bind=ctrl-z:ignore ${FZF_DEFAULT_OPTS-} ${FZF_CTRL_T_OPTS-}" $(__fzfcmd) -m "$@" | while read item; do
-    echo -n "${(q)item} "
+    echo "${(q)item} "
   done
   local ret=$?
```

---

> So .. I added smth like this.
> 
> ```shell
>    export FZF_CTRL_T_OPTS="
>    -d!
>    --with-nth 2
>    --bind \"enter:execute(for e in {+1}; do echo \$e; done)+abort\"
>    "
> ```

Consider using the additional `f` (temporary file) placeholder expression flag as an alternative to your `for` loop.

- In this example, the `print` command, which is specific to `zsh`, is used to generate a sample input for `fzf`.
  - If you were using `bash`, you might use `zsh -c 'print …'`.
```zsh
print -C3 {A..I}
# A  D  G
# B  E  H
# C  F  I
```

- Using the `{+2}` placeholder flag generates a space-separated list of the selected lines.
```zsh
print -C3 {A..I} | fzf --sync --bind 'start:select-all+become:echo {+2}'
# D E F
```

- This placeholder gives you the path to the temporary file where your selected lines are stored `{+f2}`.
```zsh
print -C3 {A..I} | fzf --sync --bind 'start:select-all+become:cat {+f2}'
# D
# E
# F
```

---

Alternatively, you can define a custom widget and assign it to a hotkey to avoid modifying the `__fsel` function.

```zsh
# zshrc
test_widget() {
	print -C3 {A..I} | fzf --sync --bind 'start:select-all+become:cat {+f2}'
}
zle -N test_widget
bindkey "^X" test_widget
```

**Reply by Shuraken007** [reply]
Thanks a lot.
